pkg/kvstore/store/store_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cilium/cilium/pkg/kvstore"
+	"github.com/cilium/cilium/pkg/testutils"
+
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testPrefix = "store-tests"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type StoreSuite struct{}
+
+type StoreEtcdSuite struct {
+	StoreSuite
+}
+
+var _ = Suite(&StoreEtcdSuite{})
+
+func (e *StoreEtcdSuite) SetUpTest(c *C) {
+	kvstore.SetupDummy("etcd")
+}
+
+func (e *StoreEtcdSuite) TearDownTest(c *C) {
+	kvstore.DeletePrefix(testPrefix)
+	kvstore.Close()
+}
+
+type StoreConsulSuite struct {
+	StoreSuite
+}
+
+var _ = Suite(&StoreConsulSuite{})
+
+func (e *StoreConsulSuite) SetUpTest(c *C) {
+	kvstore.SetupDummy("consul")
+}
+
+func (e *StoreConsulSuite) TearDownTest(c *C) {
+	kvstore.DeletePrefix(testPrefix)
+	kvstore.Close()
+}
+
+type TestType struct {
+	Name string
+
+	updated int
+	deleted int
+}
+
+var testType = TestType{}
+
+func (t *TestType) OnUpdate()          { t.updated++ }
+func (t *TestType) OnDelete()          { t.deleted++ }
+func (t *TestType) GetKeyName() string { return t.Name }
+
+func newTestType() Key {
+	t := TestType{}
+	return &t
+}
+
+func (s *StoreSuite) TestStoreCreation(c *C) {
+	// Missing Prefix must result in error
+	store, err := JoinSharedStore(Configuration{})
+	c.Assert(err, ErrorMatches, "Prefix must be specified")
```

Comment:
I've been thinking about errors, we should start define error messages this way we could easily compare the error by type and not by strings